pulsar-client/src/main/java/com/yahoo/pulsar/client/api/Consumer.java

```diff
@@ -231,4 +232,17 @@
      * breaks, the messages are redelivered after reconnect.
      */
     void redeliverUnacknowledgedMessages();
+    
+    /**
+     * Gets Consumer Stats from broker. The stats are cached for 30 seconds, if a call is made before the stats returned
+     * by the previous call expires then cached data will be returned. BrokerConsumerStats::isValid() function can be
+     * used to check if the stats are still valid.
+     * 
+     * @param partitionIndex
+     *            - optional parameter which is to be populated only if the topic is partitioned. In case of partitioned
+     *            topic pass only one PartitionIndex in a call.
+     * @return a Completeable future for BrokerConsumerStats
+     */
+    CompletableFuture<BrokerConsumerStats> getBrokerConsumerStatsAsync(int... partitionIndex);
```

Comment:
@merlimat - does this new design look good to you?

https://github.com/yahoo/pulsar/pull/254/commits/3b768355d4eba803b7cedd06e3ca0bbfa90af775